panels/CreatePromptPanel.js

```diff
@@ -0,0 +1,186 @@
+import {getUri} from "../utils/getUri";
+import {createPrompt} from "../commands";
+
+const vscode = require("vscode");
+export class CreatePromptPanel {
+ /* private _panel: any;
+  private _disposables: any[];
+  private static currentPanel: any;*/
+
+  constructor(panel, extensionUri) {
+    this._panel = panel;
+    this._disposables = []
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+    this._panel.webview.html = this._getWebviewContent(this._panel.webview, extensionUri);
+  }
+
+  static render(context, workspaceServicePrompts) {
+    if (CreatePromptPanel.currentPanel) {
+      CreatePromptPanel.currentPanel._panel.reveal(vscode.ViewColumn.One);
+    } else {
+      const panel = vscode.window.createWebviewPanel("alitacode", "Create Prompt", vscode.ViewColumn.One, {
+        enableScripts: true,
+        localResourceRoots: [vscode.Uri.joinPath(context.extensionUri, "out")]
+      });
+
+      CreatePromptPanel.currentPanel = new CreatePromptPanel(panel, context.extensionUri);
+      CreatePromptPanel.currentPanel._addReceiveMessageHandler(context, workspaceServicePrompts)
+    }
+  }
+
+  _addReceiveMessageHandler(context, workspaceServicePrompts) {
+    CreatePromptPanel.currentPanel._panel.webview.onDidReceiveMessage(
+        async message => {
+          switch (message.command) {
+            case "save":
+              await createPrompt(workspaceServicePrompts, message.promptSettings)
+              vscode.window.showInformationMessage( "Prompt saved!");
+              return;
+          }
+        },
+        undefined,
+        context.subscriptions
+    );
+  }
+
+  dispose() {
+    CreatePromptPanel.currentPanel = undefined;
+
+    this._panel.dispose();
+
+    while (this._disposables.length) {
+      const disposable = this._disposables.pop();
+      if (disposable) {
+        disposable.dispose();
+      }
+    }
+  }
+
+  _getWebviewContent(webview, extensionUri) {
+    const webviewUri = getUri(webview, extensionUri, ["out", "webview.js"]);
+    return /*html*/ `
+    <!DOCTYPE html>

```

Comment:
Can you move this huge HTML to a separate *.html file? And also create a separate script file instead of inline?

Having it in this way, it's hardly maintainable, i.e. autocomplete, linter, prettier, typo checking, syntax highlighting - all of these stuff is unavailable now 😕

Reply:
Changed, thanks for pointing, now it really is easier to maintain code